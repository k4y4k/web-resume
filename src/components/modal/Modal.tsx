import * as React from "react";
import { graphql, useStaticQuery } from "gatsby";
import tw, { css } from "twin.macro";
import BackgroundImage from "gatsby-background-image";
import { convertToBgImage } from "gbimage-bridge";
import { FiGlobe } from "@react-icons/all-files/fi/FiGlobe";
import { FiTwitter } from "@react-icons/all-files/fi/FiTwitter";
import { getImage } from "gatsby-plugin-image";

const modalStyles = css`
  p {
    ${tw`mb-2`}
  }

  a {
    ${tw`p-1 mx-1 underline cursor-pointer rounded-md`}

    &::after {
      content: ' âžœ';
    }

    &:hover {
      ${tw`text-white bg-purple-800`}
    }
  }

  span {
    ${tw`mb-2 whitespace-nowrap`}
  }

  .icon {
    ${tw`inline-block align-middle`}
    margin-bottom: 0.125rem;
  }
`;

const modalBgStyles = css`
  ${tw`fixed! top-0 z-40 flex justify-center text-center bg-purple-900`}
  height: 100vh;
  width: 100%;
`;

interface modalTypes {
  modalBg: unknown;
}

export const PureModal = ({ modalBg }: modalTypes): JSX.Element | null => {
  const [open, isOpen] = React.useState(true);

  const showModal = process.env.GATSBY_SHOW_MODAL ?? "";

  if (showModal === "") return null;

  if (!open) return null;

  return (
    <BackgroundImage
      css={modalBgStyles}
      Tag="div"
      {...modalBg}
      preserveStackingContext
    >
      <div
        css={modalStyles}
        tw="bg-white max-w-prose self-center p-6 rounded-md shadow-md"
      >
        <h1 tw="text-2xl font-bold font-mono text-center mb-4">Hi there ðŸ‘‹</h1>
        <p>
          Looks like you've found{" "}
          <a href="https://github.com/k4y4k/web-resume">k4y4k/web-resume</a>!
        </p>
        <p>
          Please note that the following resume <strong>is not mine</strong>. In
          fact, <strong>this person does not exist</strong>.
        </p>
        <p>
          All of this data has been generated with a library called{" "}
          <a href="https://github.com/marak/Faker.js/">faker.js</a>. None of it
          is real or intended to reflect or mimic a real person. Any
          similarities to people living or dead is purely coincidental.
        </p>
        <p>
          The accounts in the contact section may not exist. If they do, their
          inclusion in this project <strong>was not intentional</strong> and any
          views that their owners express are not mine. Please{" "}
          <strong>do not try to contact them</strong>. They aren't me.
        </p>
        <p>
          If you do want to contact me, you can find me on Twitter:
          <span>
            <a href="https://twitter.com/mynameis_kayak">
              <FiTwitter /> @mynameis_kayak
            </a>
          </span>
          or view my website
          <span>
            <a href="https://kayak.rocks">
              <FiGlobe /> kayak.rocks
            </a>
          </span>
          .
        </p>

        <button
          onClick={() => isOpen(false)}
          tw="hover:bg-purple-900 bg-purple-700 text-white rounded-md py-2 px-4 mx-auto block my-4"
        >
          I understand
        </button>
      </div>
    </BackgroundImage>
  );
};

const Modal = (): JSX.Element | null => {
  const { placeholderImage } = useStaticQuery(graphql`
    {
      placeholderImage: file(relativePath: { eq: "unsplash.jpg" }) {
        childImageSharp {
          gatsbyImageData(
            width: 2000
            placeholder: BLURRED
            formats: [AUTO, WEBP, AVIF]
            quality: 90
            transformOptions: {
              duotone: { shadow: "#000000", highlight: "#ffffff", opacity: 50 }
              rotate: 180
            }
          )
        }
      }
    }
  `);

  const modalBg = convertToBgImage(getImage(placeholderImage));

  return <PureModal modalBg={modalBg} />;
};

export default Modal;
