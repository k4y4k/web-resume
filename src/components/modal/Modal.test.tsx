/* eslint-disable @typescript-eslint/no-var-requires */
import * as React from "react";
import { render, screen } from "@testing-library/react";
import { PureModal } from "./Modal";

jest.mock("short-uuid");

describe("<PureModal />", () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...OLD_ENV };
  });

  afterAll(() => {
    process.env = OLD_ENV;
  });

  const data = {
    fluid: {
      src: "/static/55019b471591030c02c094b138609888/1a09b/unsplash.jpg",
      srcSet:
        "/static/55019b471591030c02c094b138609888/c4ac9/unsplash.jpg 250w,\n/static/55019b471591030c02c094b138609888/3d684/unsplash.jpg 500w,\n/static/55019b471591030c02c094b138609888/1a09b/unsplash.jpg 1000w",
      sizes: "(min-width: 1000px) 1000px, 100vw",
      srcSetAvif:
        "/static/55019b471591030c02c094b138609888/b4a16/unsplash.avif 250w,\n/static/55019b471591030c02c094b138609888/819e2/unsplash.avif 500w,\n/static/55019b471591030c02c094b138609888/97459/unsplash.avif 1000w",
      srcAvif: "",
      srcSetWebp:
        "/static/55019b471591030c02c094b138609888/7a308/unsplash.webp 250w,\n/static/55019b471591030c02c094b138609888/22b76/unsplash.webp 500w,\n/static/55019b471591030c02c094b138609888/00280/unsplash.webp 1000w",
      srcWebp: "",
      base64:
        "data:image/jpeg;base64,/9j/2wBDAAMCAgMCAgMDAwMEAwMEBQgFBQQEBQoHBwYIDAoMDAsKCwsNDhIQDQ4RDgsLEBYQERMUFRUVDA8XGBYUGBIUFRT/2wBDAQMEBAUEBQkFBQkUDQsNFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBT/wgARCAAeABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAYEBQcD/8QAFgEBAQEAAAAAAAAAAAAAAAAAAgQF/9oADAMBAAIQAxAAAAFLjZ31VWhFiSaKei6YlX5EcrAj/8QAHRAAAQQDAQEAAAAAAAAAAAAAAwABAgQFEhQRJP/aAAgBAQABBQIVvrFSMOxP60YugaF0lcj6xWRx7HZ20n0tNWRuImSD4Rf/xAAaEQEAAQUAAAAAAAAAAAAAAAABAAIDEBEh/9oACAEDAQE/AS8nIBqVY//EABkRAQEBAAMAAAAAAAAAAAAAAAEAAhESQf/aAAgBAgEBPwFyM82j27N//8QAIBAAAQMDBQEAAAAAAAAAAAAAAgABAxEhMRASIkFhUf/aAAgBAQAGPwKUnjCMLs91PHTcEd2cHyuZsfx6KSP6d3QsB7G79XLPiKeHPY170qWU167k5+00/8QAHhAAAgMAAgMBAAAAAAAAAAAAAREAITFBYVFxgbH/2gAIAQEAAT8hpqJWPsIcCMYYR/RkLG84Rsd1DARZi/x1PWqjaYzPJ1cEJ8gjlENJHI4dx5h9yUca8iCB9/dOKf/aAAwDAQACAAMAAAAQV+MA/8QAGxEBAAICAwAAAAAAAAAAAAAAAQARIUGBsfD/2gAIAQMBAT8QLvk31Ehp9zBsls//xAAbEQEBAAEFAAAAAAAAAAAAAAABABEhMVGB8P/aAAgBAgEBPxBvhkFNPdRbLHf/xAAfEAEAAgICAgMAAAAAAAAAAAABESEAMUFxUWGBkfD/2gAIAQEAAT8QqC4BqCCWmxn3g+AsHWPGg4Md43QJbyqp4GVq4omsTqhDbgenfjJwDFDKakdh+tySPIC0QkyksMzhD5ChYZCRak9/OMlr1fCMx94ORmeVcYib2EkdC4e5nEFCg8pga9ViRz//2Q==",
      aspectRatio: 0.6666666666666666,
    },
  };

  test("does not render if GATSBY_SHOW_MODAL is false", () => {
    const uuid = require("short-uuid");
    uuid.generate.mockImplementation(() => "73WakrfVbNJBaAmhQtEeDv");

    const { container } = render(<PureModal modalBg={data} />);

    expect(container).toMatchSnapshot();
  });

  test("does not render if GATSBY_SHOW_MODAL is ''", () => {
    process.env.GATSBY_SHOW_MODAL = "";
    const uuid = require("short-uuid");
    uuid.generate.mockImplementation(() => "73WakrfVbNJBaAmhQtEeDv");

    const { container } = render(<PureModal modalBg={data} />);

    expect(container).toMatchSnapshot();
  });

  test("renders OK otherwise", () => {
    process.env.GATSBY_SHOW_MODAL = "true";

    const uuid = require("short-uuid");
    uuid.generate.mockImplementation(() => "73WakrfVbNJBaAmhQtEeDv");

    const { container } = render(<PureModal modalBg={data} />);

    expect(container).toMatchSnapshot();
  });

  test("button makes it go away ", () => {
    process.env.GATSBY_SHOW_MODAL = "true";

    const uuid = require("short-uuid");
    uuid.generate.mockImplementation(() => "73WakrfVbNJBaAmhQtEeDv");

    const { container } = render(<PureModal modalBg={data} />);

    screen.getByText("I understand", { exact: false }).click();

    expect(container).toMatchSnapshot();
  });
});
