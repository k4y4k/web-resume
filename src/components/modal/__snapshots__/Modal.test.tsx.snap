// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PureModal /> button makes it go away  1`] = `<div />`;

exports[`<PureModal /> does not render if GATSBY_SHOW_MODAL is '' 1`] = `<div />`;

exports[`<PureModal /> does not render if GATSBY_SHOW_MODAL is false 1`] = `<div />`;

exports[`<PureModal /> renders OK otherwise 1`] = `
.c0 {
  position: fixed !important;
  top: 0px;
  z-index: 40;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  --tw-bg-opacity: 1;
  background-color: rgba(76,29,149,var(--tw-bg-opacity));
  height: 100vh;
  width: 100%;
}

.c1 {
  --tw-bg-opacity: 1;
  background-color: rgba(255,255,255,var(--tw-bg-opacity));
  max-width: 65ch;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 1.5rem;
  border-radius: 0.375rem;
  --tw-shadow: 0 4px 6px -1px rgba(0,0,0,0.1),0 2px 4px -1px rgba(0,0,0,0.06);
  box-shadow: var(--tw-ring-offset-shadow,0 0 #0000),var(--tw-ring-shadow,0 0 #0000),var(--tw-shadow);
}

.c1 p {
  margin-bottom: 0.5rem;
}

.c1 a {
  padding: 0.25rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  border-radius: 0.375rem;
}

.c1 a::after {
  content: ' âžœ';
}

.c1 a:hover {
  --tw-text-opacity: 1;
  color: rgba(255,255,255,var(--tw-text-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(91,33,182,var(--tw-bg-opacity));
}

.c1 span {
  margin-bottom: 0.5rem;
  white-space: nowrap;
}

.c1 .icon {
  display: inline-block;
  vertical-align: middle;
  margin-bottom: 0.125rem;
}

.c2 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-weight: 700;
  font-family: Inconsolata,ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;
  text-align: center;
  margin-bottom: 1rem;
}

.c3 {
  --tw-bg-opacity: 1;
  background-color: rgba(109,40,217,var(--tw-bg-opacity));
  --tw-text-opacity: 1;
  color: rgba(255,255,255,var(--tw-text-opacity));
  border-radius: 0.375rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  margin-left: auto;
  margin-right: auto;
  display: block;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c3:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(76,29,149,var(--tw-bg-opacity));
}

<div>
  <div
    class="c0 gbi-1850775931-undefined"
    style="background-position: center; background-repeat: no-repeat; background-size: cover; position: relative;"
  >
    <style>
      
          .Modal___StyledBackgroundImage-wzs0mx-0.c0.gbi-1850775931-undefined:before,
          .Modal___StyledBackgroundImage-wzs0mx-0.c0.gbi-1850775931-undefined:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;
background-color: rgba(76,29,149,var(--tw-bg-opacity));

          }
          .Modal___StyledBackgroundImage-wzs0mx-0.c0.gbi-1850775931-undefined:before {
            z-index: -100;
            background-image: url(data:image/jpeg;base64,/9j/2wBDAAMCAgMCAgMDAwMEAwMEBQgFBQQEBQoHBwYIDAoMDAsKCwsNDhIQDQ4RDgsLEBYQERMUFRUVDA8XGBYUGBIUFRT/2wBDAQMEBAUEBQkFBQkUDQsNFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBT/wgARCAAeABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAYEBQcD/8QAFgEBAQEAAAAAAAAAAAAAAAAAAgQF/9oADAMBAAIQAxAAAAFLjZ31VWhFiSaKei6YlX5EcrAj/8QAHRAAAQQDAQEAAAAAAAAAAAAAAwABAgQFEhQRJP/aAAgBAQABBQIVvrFSMOxP60YugaF0lcj6xWRx7HZ20n0tNWRuImSD4Rf/xAAaEQEAAQUAAAAAAAAAAAAAAAABAAIDEBEh/9oACAEDAQE/AS8nIBqVY//EABkRAQEBAAMAAAAAAAAAAAAAAAEAAhESQf/aAAgBAgEBPwFyM82j27N//8QAIBAAAQMDBQEAAAAAAAAAAAAAAgABAxEhMRASIkFhUf/aAAgBAQAGPwKUnjCMLs91PHTcEd2cHyuZsfx6KSP6d3QsB7G79XLPiKeHPY170qWU167k5+00/8QAHhAAAgMAAgMBAAAAAAAAAAAAAREAITFBYVFxgbH/2gAIAQEAAT8hpqJWPsIcCMYYR/RkLG84Rsd1DARZi/x1PWqjaYzPJ1cEJ8gjlENJHI4dx5h9yUca8iCB9/dOKf/aAAwDAQACAAMAAAAQV+MA/8QAGxEBAAICAwAAAAAAAAAAAAAAAQARIUGBsfD/2gAIAQMBAT8QLvk31Ehp9zBsls//xAAbEQEBAAEFAAAAAAAAAAAAAAABABEhMVGB8P/aAAgBAgEBPxBvhkFNPdRbLHf/xAAfEAEAAgICAgMAAAAAAAAAAAABESEAMUFxUWGBkfD/2gAIAQEAAT8QqC4BqCCWmxn3g+AsHWPGg4Md43QJbyqp4GVq4omsTqhDbgenfjJwDFDKakdh+tySPIC0QkyksMzhD5ChYZCRak9/OMlr1fCMx94ORmeVcYib2EkdC4e5nEFCg8pga9ViRz//2Q==);
            
            opacity: 1; 
          }
          .Modal___StyledBackgroundImage-wzs0mx-0.c0.gbi-1850775931-undefined:after {
            z-index: -101;
            
            background-image: url(data:image/jpeg;base64,/9j/2wBDAAMCAgMCAgMDAwMEAwMEBQgFBQQEBQoHBwYIDAoMDAsKCwsNDhIQDQ4RDgsLEBYQERMUFRUVDA8XGBYUGBIUFRT/2wBDAQMEBAUEBQkFBQkUDQsNFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBT/wgARCAAeABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAYEBQcD/8QAFgEBAQEAAAAAAAAAAAAAAAAAAgQF/9oADAMBAAIQAxAAAAFLjZ31VWhFiSaKei6YlX5EcrAj/8QAHRAAAQQDAQEAAAAAAAAAAAAAAwABAgQFEhQRJP/aAAgBAQABBQIVvrFSMOxP60YugaF0lcj6xWRx7HZ20n0tNWRuImSD4Rf/xAAaEQEAAQUAAAAAAAAAAAAAAAABAAIDEBEh/9oACAEDAQE/AS8nIBqVY//EABkRAQEBAAMAAAAAAAAAAAAAAAEAAhESQf/aAAgBAgEBPwFyM82j27N//8QAIBAAAQMDBQEAAAAAAAAAAAAAAgABAxEhMRASIkFhUf/aAAgBAQAGPwKUnjCMLs91PHTcEd2cHyuZsfx6KSP6d3QsB7G79XLPiKeHPY170qWU167k5+00/8QAHhAAAgMAAgMBAAAAAAAAAAAAAREAITFBYVFxgbH/2gAIAQEAAT8hpqJWPsIcCMYYR/RkLG84Rsd1DARZi/x1PWqjaYzPJ1cEJ8gjlENJHI4dx5h9yUca8iCB9/dOKf/aAAwDAQACAAMAAAAQV+MA/8QAGxEBAAICAwAAAAAAAAAAAAAAAQARIUGBsfD/2gAIAQMBAT8QLvk31Ehp9zBsls//xAAbEQEBAAEFAAAAAAAAAAAAAAABABEhMVGB8P/aAAgBAgEBPxBvhkFNPdRbLHf/xAAfEAEAAgICAgMAAAAAAAAAAAABESEAMUFxUWGBkfD/2gAIAQEAAT8QqC4BqCCWmxn3g+AsHWPGg4Md43QJbyqp4GVq4omsTqhDbgenfjJwDFDKakdh+tySPIC0QkyksMzhD5ChYZCRak9/OMlr1fCMx94ORmeVcYib2EkdC4e5nEFCg8pga9ViRz//2Q==);
            
          }
        
    </style>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Hi there ðŸ‘‹
      </h1>
      <p>
        Looks like you've found
         
        <a
          href="https://github.com/k4y4k/web-resume"
        >
          k4y4k/web-resume
        </a>
        !
      </p>
      <p>
        Please note that the following resume 
        <strong>
          is not mine
        </strong>
        . In fact, 
        <strong>
          this person does not exist
        </strong>
        .
      </p>
      <p>
        All of this data has been generated with a library called
         
        <a
          href="https://github.com/marak/Faker.js/"
        >
          faker.js
        </a>
        . None of it is real or intended to reflect or mimic a real person. Any similarities to people living or dead is purely coincidental.
      </p>
      <p>
        The accounts in the contact section may not exist. If they do, their inclusion in this project 
        <strong>
          was not intentional
        </strong>
         and any views that their owners express are not mine. Please
         
        <strong>
          do not try to contact them
        </strong>
        . They aren't me.
      </p>
      <p>
        If you do want to contact me, you can find me on Twitter:
        <span>
          <a
            href="https://twitter.com/mynameis_kayak"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
              />
            </svg>
             @mynameis_kayak
          </a>
        </span>
        or view my website
        <span>
          <a
            href="https://kayak.rocks"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="2"
                x2="22"
                y1="12"
                y2="12"
              />
              <path
                d="M12 2a15.3 15.3 0 0 1 4 10 15.3 15.3 0 0 1-4 10 15.3 15.3 0 0 1-4-10 15.3 15.3 0 0 1 4-10z"
              />
            </svg>
             kayak.rocks
          </a>
        </span>
        .
      </p>
      <button
        class="c3"
      >
        I understand
      </button>
    </div>
  </div>
</div>
`;
